### PR TITLE
Added get balance controller and API request filter

### PR DESCRIPTION
**Pull request by WarFire5**

### CODE REVIEW

**Comment by Myslyaev** [comment]
Hey @WarFire5, while reviewing your PR, I'd suggest the following code changes:

👉 [**Code Suggestion for #2**](https://gitkraken.dev/drafts/ba4db7ba-b3c5-4556-9d64-3ba3b95e8fa6?source=PRComment)
>https://github.com/WarFire5/TransactionStore/pull/2

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/ba4db7ba-b3c5-4556-9d64-3ba3b95e8fa6?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPTDliTDFNNXdQRHBwIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

File: TransactionStore.Business/Services/TransactionsService.cs [hunk 1]
```diff
@@ -14,11 +16,25 @@ public class TransactionsService : ITransactionsService
...
     {
         _transactionsRepository = usersRepository;
         _mapper = mapper;
         _addTransactionValidator = addTransactionValidator;
     }
+
+    public AccountBalanceResponse GetBalanceByAccountId(Guid id)
+    {
+        _logger.Information("Вызываем метод репозитория");
+        List<TransactionDto> transactionDtos = _transactionsRepository.GetBalanceByAccountId(id);
```
Comment (Negodyaj) [inline TransactionStore.Business/Services/TransactionsService.cs]: GetTransactionsByAccountId

File: TransactionStore.Business/Services/TransactionsService.cs [hunk 2]
```diff
@@ -14,11 +16,25 @@ public class TransactionsService : ITransactionsService
...
+        var balance = transactionDtos.Sum(t => t.Amount);
+
+        _logger.Information("Считаем и передаем баланс");
+        TransactionDto accountBalance = new TransactionDto();
+        accountBalance.Amount = balance;
```
Comment (Negodyaj) [inline TransactionStore.Business/Services/TransactionsService.cs]: {

File: TransactionStore.Business/Services/TransactionsService.cs [hunk 3]
```diff
@@ -14,11 +16,25 @@ public class TransactionsService : ITransactionsService
...
+        accountBalance.AccountId = transactionDtos[0].AccountId;
+        accountBalance.CurrencyType = transactionDtos[0].CurrencyType;
```
Comment (Negodyaj) [inline TransactionStore.Business/Services/TransactionsService.cs]: }

File: TransactionStore.Business/Services/TransactionsService.cs [hunk 4]
```diff
@@ -14,11 +16,25 @@ public class TransactionsService : ITransactionsService
...
     }
+
+    public AccountBalanceResponse GetBalanceByAccountId(Guid id)
+    {
+        _logger.Information("Вызываем метод репозитория");
+        List<TransactionDto> transactionDtos = _transactionsRepository.GetBalanceByAccountId(id);
+        var balance = transactionDtos.Sum(t => t.Amount);
+
+        _logger.Information("Считаем и передаем баланс");
+        TransactionDto accountBalance = new TransactionDto();
```
Comment (Negodyaj) [inline TransactionStore.Business/Services/TransactionsService.cs]: change to Response

File: TransactionStore.Core/Models/Transactions/TransactionsMappingProfile.cs [hunk 5]
```diff
@@ -1,10 +1,14 @@
...
+using TransactionStore.Core.Models.Transactions.Responses;
 
 namespace TransactionStore.Core.Models.Transactions;
 
 public class TransactionsMappingProfile : Profile
 {
     public TransactionsMappingProfile()
     {
+        CreateMap<TransactionDto, AccountBalanceResponse>()
+            .ForMember(d => d.Balance, opt => opt.MapFrom(s =>s.Amount));
```
Comment (Negodyaj) [inline TransactionStore.Core/Models/Transactions/TransactionsMappingProfile.cs:12]: тебе не нужен такой маппинг)

**Pull request merged**